packages/storage/src/ref.ts

```diff
@@ -40,12 +40,12 @@ interface ReferenceStoreOptions {
 
 /** @experimental */
 class ReferenceStore implements AsyncReadable<RequestInit> {
-	#refs: Map<string, ReferenceEntry>;
+	#refs: Promise<Map<string, ReferenceEntry>>;
 	#opts: ReferenceStoreOptions;
 	#overrides: RequestInit;
 
 	constructor(
-		refs: Map<string, ReferenceEntry>,
+		refs: Promise<Map<string, ReferenceEntry>>,
```

Comment:
Maybe we could make this `Promise<Map<string, ReferenceEntry>> | Map<string, ReferencEntry>`?

That way we support the previous constructor as well.